services/mlmodel/tflite_cpu/tflite_cpu_test.go

```diff
@@ -0,0 +1,156 @@
+	modelLoc2 := artifact.MustPath("vision/tflite/effnet0.tflite")
+	modelLoc3 := artifact.MustPath("vision/tflite/mobileBERT.tflite")
+	emptyCfg := TFLiteConfig{} // empty config
+	cfg := TFLiteConfig{       // detector config
+		ModelPath:  modelLoc,
+		NumThreads: 2,
+	}
+	cfg2 := TFLiteConfig{ // classifier config
+		ModelPath:  modelLoc2,
+		NumThreads: 2,
+	}
+	cfg3 := TFLiteConfig{ // text classifier config
+		ModelPath:  modelLoc3,
+		NumThreads: 1,
+	}
+
+	// Test that empty config gives error about loading model
+	emptyGot, err := CreateTFLiteCPUModel(ctx, &emptyCfg)
+	test.That(t, emptyGot, test.ShouldResemble, &Model{})
+	test.That(t, err.Error(), test.ShouldContainSubstring, "could not add model")
+	test.That(t, cfg, test.ShouldNotBeNil)
+
+	// Test that a detector would give the expected output on the dog image
+	// Creating the model should populate model and attrs, but not metadata
+	got, err := CreateTFLiteCPUModel(ctx, &cfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, got.model, test.ShouldNotBeNil)
+	test.That(t, got.attrs, test.ShouldNotBeNil)
+	test.That(t, got.metadata, test.ShouldBeNil)
+
+	// Test that the Metadata() works on detector
+	gotMD, err := got.Metadata(ctx)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, gotMD, test.ShouldNotBeNil)
+	test.That(t, got.metadata, test.ShouldNotBeNil)
+
+	test.That(t, gotMD.Inputs[0].Name, test.ShouldResemble, "image")
+	test.That(t, gotMD.Outputs[0].Name, test.ShouldResemble, "location")
+	test.That(t, gotMD.Outputs[1].Name, test.ShouldResemble, "category")
+	test.That(t, gotMD.Outputs[2].Name, test.ShouldResemble, "score")
+	test.That(t, gotMD.Inputs[0].DataType, test.ShouldResemble, "uint8")
+	test.That(t, gotMD.Outputs[0].DataType, test.ShouldResemble, "float32")
+	test.That(t, gotMD.Outputs[1].AssociatedFiles[0].Name, test.ShouldResemble, "labelmap.txt")
+
+	// Test that the Infer() works on detector
+	pic, err := rimage.NewImageFromFile(artifact.MustPath("vision/tflite/dogscute.jpeg"))
+	test.That(t, err, test.ShouldBeNil)
+	resized := resize.Resize(uint(got.metadata.Inputs[0].Shape[1]), uint(got.metadata.Inputs[0].Shape[2]), pic, resize.Bilinear)
+	imgBytes := rimage.ImageToUInt8Buffer(resized)
+	test.That(t, imgBytes, test.ShouldNotBeNil)
+	inputMap := make(map[string]interface{})
+	inputMap["image"] = imgBytes
+
+	gotOutput, err := got.Infer(ctx, inputMap)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, gotOutput, test.ShouldNotBeNil)
+
+	test.That(t, gotOutput["number of detections"], test.ShouldResemble, []float32{25})
+	test.That(t, len(gotOutput["score"].([]float32)), test.ShouldResemble, 25)
+	test.That(t, len(gotOutput["location"].([]float32)), test.ShouldResemble, 100)
+	test.That(t, len(gotOutput["category"].([]float32)), test.ShouldResemble, 25)
+	test.That(t, gotOutput["category"].([]float32)[0], test.ShouldEqual, 17) // 17 is dog
+
+	// Test that the tflite classifier gives the expected output on the lion image
```

Comment:
Turn each different model you are testing out into separate tests. When it's one big test, the failure prevents the rest of the test from running.

Reply:
Done